test/bench.hs

```diff
@@ -101,5 +115,11 @@ main = tinfoilBench [
                              ]
   , env ((,) <$> generate' arbitrary <*> generate' arbitrary) $ \ ~(sk, bs) ->
       bgroup "mac/hmacSHA256" $ [ bench "hmacSHA256" $ nf (hmacSHA256 sk) bs
-                             ]
+                                ]
+  , env genEd25519 $ \ ~(sk, pk, sig, msg) ->
+      bgroup "signing/ed25519" $ [
```

Comment:
Are you using `StrictData` (the `~` character) here or something? Is that going to compile with ghc-7.8?